pkg/kotsadm/objects/scripts/export-minio-data.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+set -e
+
+# This script exports configuration, bucket metadata, iam settings, and bucket content from the old minio instance
+# to the shared migration directory
+
+# check if the migration has already been completed
+if [ -f /export/.migration ];
+then
+    echo "migration already completed, no-op"
+    exit 0
+fi
+
+# validate environment variables
+if [ -z $KOTSADM_MINIO_MIGRATION_WORK_DIR ] ||
+   [ -z $KOTSADM_MINIO_LEGACY_ALIAS ] ||
+   [ -z $KOTSADM_MINIO_ENDPOINT ] ||
+   [ -z $MINIO_ACCESS_KEY ] ||
+   [ -z $MINIO_SECRET_KEY ] ||
+   [ -z $KOTSADM_MINIO_BUCKET_NAME ];
+then
+    echo 's3 migration environment variables not set'
+    exit 1
+fi
+
+# check if the migration has already been completed
+if [ -f /export/.migration ];
+then
+    echo "migration already completed, no-op"
+    exit 0
+fi
+
+# change the working directory to the migration directory
+cd $KOTSADM_MINIO_MIGRATION_WORK_DIR
+
+echo "starting minio instance"
+/bin/sh -ce "/usr/bin/docker-entrypoint.sh minio -C /home/minio/.minio/ server /export" &
+MINIO_PID=$!
+
+# wait for minio to be ready
+until curl -s $KOTSADM_MINIO_ENDPOINT/minio/health/ready; do
+    echo "waiting for minio to be ready"
+    sleep 1
+done
+
+# alias the minio instance
+echo "aliasing minio instance"
+$KOTSADM_MINIO_MIGRATION_WORK_DIR/bin/mc alias set $KOTSADM_MINIO_LEGACY_ALIAS $KOTSADM_MINIO_ENDPOINT $MINIO_ACCESS_KEY $MINIO_SECRET_KEY
+
+echo "exporting minio configuration"
+$KOTSADM_MINIO_MIGRATION_WORK_DIR/bin/mc admin config export $KOTSADM_MINIO_LEGACY_ALIAS > $KOTSADM_MINIO_MIGRATION_WORK_DIR/config.txt
+
+echo "exporting minio bucket metadata"
+set +e
+$KOTSADM_MINIO_MIGRATION_WORK_DIR/bin/mc admin cluster bucket export $KOTSADM_MINIO_LEGACY_ALIAS
+# old instances may throw error: "This 'admin' API is not supported by server in 'mode-server-fs'"
+# ref: https://github.com/minio/mc/issues/4409 and https://github.com/minio/console/issues/863
+if [ $? -ne 0 ]; then
+    echo "exporting minio bucket metadata failed, skipping"
+fi
+
+echo "exporting minio iam info"
+$KOTSADM_MINIO_MIGRATION_WORK_DIR/bin/mc admin cluster iam export $KOTSADM_MINIO_LEGACY_ALIAS
+# old instances may throw error: "This 'admin' API is not supported by server in 'mode-server-fs'"
+# ref: https://github.com/minio/mc/issues/4409 and https://github.com/minio/console/issues/863
+if [ $? -ne 0 ]; then
+    echo "exporting minio iam info failed, skipping"
+fi
+set -e
```

Comment:
I've seen these fail with older KOTS versions using old MinIO images.  Even if we don't migrate these, we are still able to create the bucket in the new instance and migrate the data without any lost functionality in KOTS.

Reply:
this is now outdated.  we're not migrating config or metadata, only the bucket contents